x-pack/plugin/core/src/main/java/org/elasticsearch/license/ClusterStateLicenseService.java

```diff
@@ -464,49 +462,20 @@ public void clusterChanged(ClusterChangedEvent event) {
         }
     }
 
-    protected String getExpiryWarning(long licenseExpiryDate, long currentTime) {
-        final long diff = licenseExpiryDate - currentTime;
-        if (LicenseSettings.LICENSE_EXPIRATION_WARNING_PERIOD.getMillis() > diff) {
-            final long days = TimeUnit.MILLISECONDS.toDays(diff);
-            final String expiryMessage = (days == 0 && diff > 0)
-                ? "expires today"
-                : (diff > 0
-                    ? String.format(Locale.ROOT, "will expire in [%d] days", days)
-                    : String.format(
-                        Locale.ROOT,
-                        "expired on [%s]",
-                        ClusterStateLicenseService.DATE_FORMATTER.formatMillis(licenseExpiryDate)
-                    ));
-            return "Your license "
-                + expiryMessage
-                + ". "
-                + "Contact your administrator or update your license for continued use of features";
-        }
-        return null;
-    }
-
     private void updateXPackLicenseState(License license) {
-        long time = clock.millis();
         if (license == LicensesMetadata.LICENSE_TOMBSTONE) {
             // implies license has been explicitly deleted
-            xPacklicenseState.update(License.OperationMode.MISSING, false, getExpiryWarning(LicenseUtils.getExpiryDate(license), time));
+            xPacklicenseState.update(LicenseUtils.getXPackLicenseStatus(license, clock));
```

Comment:
It could be functionally unnecessary to keep the check, but technically there is a different handling between null and tombstone and think handling that nuance is bit beyond the refactoring done in this PR. I can address it in a follow up.